internal/ethapi/api.go

```diff
@@ -1323,6 +1323,17 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 		}
 		return result.Failed(), result, nil
 	}
+
+	// When calling across contracts, an opCall call error out of gas will occur. However, this error is not execution reverted,
+	// causing the error and cannot be captured by the call stack. Therefore, when subsequent opcode is executed, opRevert is
+	// triggered because the return value is not the expected value. But if this error is triggered during the bisection process,
+	// it can be tolerated because smaller gas is used in the estimation.
+	//
+	// Therefore, the function of this flag is that as long as estimateGas is successful once during the bisection process, it will
+	// not terminate. The process of bisection and throwing an evm error (drawing on the idea of ​​Ethereum, using the largest balance
+	// to estimate once, if successful, no subsequent judgment errors will be made, if failed, an exception will be thrown directly)
+	var success bool
+
 	// Execute the binary search and hone in on an executable gas limit
 	for lo+1 < hi {
 		mid := (hi + lo) / 2
```

Comment:
if do call error with first mid gas, this PR don't fix the problem
it is better to do call with hi gas value before into binary search